src/main/java/CodeBuildException.java

```diff
@@ -0,0 +1,14 @@
+import org.jenkinsci.plugins.scriptsecurity.sandbox.whitelists.Whitelisted;
+
+public class CodeBuildException extends InterruptedException{
+    private final CodeBuildResult codeBuildResult;
+
+    public CodeBuildException(CodeBuildResult codeBuildResult) {
+        this.codeBuildResult = codeBuildResult;
+    }
+
+    @Whitelisted
```

Comment:
We need to Whitelist some fields, otherwise the ScriptSecurity Plugin blocks some function calls. Since this data is not exactly secret, it is okay to WhiteList it in the plugin, rather than users having to approve it from Jenkins everytime.